target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -376,7 +477,22 @@ module pyth::pyth {
     ///
     /// Please read more information about the update fee here: https://docs.pyth.network/consume-data/on-demand#fees
     public fun get_update_fee(update_data: &vector<vector<u8>>): u64 {
-        state::get_base_update_fee() * vector::length(update_data)
+        let i = 0;
+        let total_updates = 0;
+        while (i < vector::length(update_data)) {
+            let cur = cursor::init(*vector::borrow(update_data, i));
+            let header: u64 = deserialize::deserialize_u32(&mut cur);
+            if (header == PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
+                let updates = parse_accumulator(&mut cur);
```

Comment:
leave a TODO that this may be expensive and need to be optimized. (this is a general annoying thing about programming on the blockchain, which is that you have to be very sensitive to computation cost. which often forces you to implement your code in a less obvious way)